### PR TITLE
Completed Capstone Project

### PR DESCRIPTION
**Pull request by Bettenoch**
In this, project @Graciano1997 and @Bettenoch did the following tasks:

- Set up Gitflow
- Set up webpack
- Set up a JavaScript testing library (Jest)
- Choose images, text, icons, fonts.
- Choose external  API (https://www.tvmaze.com/api)
- Add header and footer pages.
- Create the main part of the homepage that keeps the layout from the wireframe:
- Add popup function to create modal dynamically and display movie information.
- Add counter function to check the number of like  and comments items in a container.
- Display the number of likes for each movie.
- Display the number of comments for each movies on the popup section.
- Deployed Application
- Add responsivity to the website
- Make sure that function for the items counter is covered with unit tests using Jest. Make sure that you have tests for all edge cases.
- Fixed linters errors
- Updated readme (Add video presentation link)

Thank you for reviewing our pull requests 🙏 🙏

### CODE REVIEW

File: src/modules/likeController.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+const likesReader = async () => {
+  await fetch(`${Variable.involvementApiUrl}likes/`)
+    .then((response) => response.json())
+    .then((json) => {
+      document.querySelectorAll('.likes-number').forEach((itemLike, index) => {
+        itemLike.textContent = (json[index].likes > 1) ? (`${json[index].likes} Likes`) : (`${json[index].likes} Like`);
+      });
+    });
+};
```
Comment (muneebulrehman) [inline src/modules/likeController.js:21]: As per the project requirements, when a user clicks on the Like button of an item, the interaction should get recorded in the involvement API, and the screen gets updated. However, in your case, the former happens but the latter doesn't take effect immediately. I either have to refresh the page or click the like button again to see the number of like getting updated. But when I click the like button again, it displays the like number of previous clicked event. And when I refresh the page, it displays the correct like number. Kindly fix this error.

https://github.com/Graciano1997/js_moduleII_capstone/assets/33268927/9e8df8e5-6898-4f99-91ba-941b921310c7

File: src/index.html [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+                        </div>
+                        <div class="moovie-item-image-description">
+                            <h4 class="item-name">Name</h4>
+                            <div class="moovie-item-likes">
+                                <p class="heart"><i class="fa-regular fa-heart" id="likeIcon"></i></p>
+                                <input type="hidden" name="idmoovie" value="" class="moovieId">
+                                  <p class="likes-number"></p>
+                            </div>
+                        </div>
+                    </div>
```
Comment (muneebulrehman) [inline src/index.html:195]: This whole section should be generated dynamically using JS. However, in your case, you have written the HTML manually which makes the concept of a dynamic website pointless. Imagine if you have 20 or 30 movies. The HTML would be very long in that case, which is not a point of this module. Kindly remove this hard-coded HTML and use JS to generate the code.

File: src/index.html [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+            <button type="button"  id="btnSee"><i class="fa-solid fa-eye-slash"></i> See Less</button>
+        </div>
+    </main>
+    
+    <section class="popup-container hide">
+    </section>
+    
+    <footer>
+            <p>Created by Microverse students under MIT Licence</p>
+    </footer>
```
Comment (muneebulrehman) [inline src/index.html:214]: The footer should be separated from the main body by a border as in the provided wireframe design. You don't need to go crazy with the styling but your design should follow the provided design.

| Your design  | Provided design |
| ------------- | ------------- |
| image | image |

**Review by muneebulrehman — CHANGES_REQUESTED** [review]
Hi team, 

Good job so far! I see you put a lot of effort into this project. However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- No linter errors
- Descriptive PR
- Used ES6 features
- Clean code

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/likeController.js (hunk 1 above)
Comment (Graciano1997) [reply src/modules/likeController.js:21]: hello @muneebulrehman  .

**Review by iambenkis — APPROVED** [review]
## Hi @Bettenoch and @Graciano1997   ,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
### Highlights
- Descriptive pull request message with screenshots ✔️
- Well organized project ✔️
- No linter errors ✔️
- Descriptive PR ✔️
- Used ES6 features ✔️
- Clean code ✔️
approval


# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me  @iambenkis in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
